FS check on boot partition

agherzan opened the pull request:
Fixes #816

---
### Contributor checklist
<!-- For completed items, change [ ] to [x].  -->
- [x] Changes have been tested
- [x] `Change-type` present on at least one commit
- [x] `Signed-off-by` is present
- [x] The PR complies with the [Open Embedded Commit Patch Message Guidelines](http://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines)
<!-- optional: `Changelog-entry` present on at least one commit if you want to set the changelog entry manually-->

### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)

agherzan commented:
@floion it not only removes it but renames it to make it non ext4 specific. Fsckext4 - > fsck. That's what I mean by not ext4 specific.

floion commented:
fsckext4 was a initramfs script we added. You commit removed it completely but also says it makes it (fsckext4) non ext4 specific. fsckext4 is not fsck.ext4, they are different things. What I don't understand is why fsckext4 is said to be made non ext4 specific if it's removed altogether.

agherzan commented:
I'm only about the initramfs module fsckext4. It used to have a name and contain only fscks for ext4. I renamed it to not be ext4 specific anymore and added boot checks too (vfat) making it a module that does fsck on any partition type. Does this clarify it?

agherzan commented:
I'm only about the initramfs module fsckext4. It used to have a name and contain only fscks for ext4. I renamed it to not be ext4 specific anymore and added boot checks too (vfat) making it a module that does fsck on any partition type. Does this clarify it, @floion ?

floion commented:
I understand the changes. I'm just talking about the wording in the commit log.

floion commented:
@resin-jenkins retest this please

floion commented:
@resin-jenkins retest this please

agherzan commented:
Depends on https://github.com/resin-os/meta-resin/pull/972 . Will be rebased when that gets merged.

agherzan commented:
Rebased on top of current master.
